infra/spawn_data_pipeline.py

```diff
@@ -0,0 +1,124 @@
+#!/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright 2019 Mozilla
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This script triggers the data pipeline for the bugbug project
+"""
+
+from __future__ import print_function
+
+import argparse
+import os
+import sys
+
+import jsone
+import requests.packages.urllib3
+import taskcluster
+import yaml
+
+requests.packages.urllib3.disable_warnings()
+
+TASKCLUSTER_DEFAULT_URL = "https://taskcluster.net"
+
+
+def get_taskcluster_options():
+    """
+    Helper to get the Taskcluster setup options
+    according to current environment (local or Taskcluster)
+    """
+    options = taskcluster.optionsFromEnvironment()
+    proxy_url = os.environ.get("TASKCLUSTER_PROXY_URL")
+
+    if proxy_url is not None:
+        # Always use proxy url when available
+        options["rootUrl"] = proxy_url
+
+    if "rootUrl" not in options:
+        # Always have a value in root url
+        options["rootUrl"] = TASKCLUSTER_DEFAULT_URL
+
+    return options
+
+
+def main():
+    parser = argparse.ArgumentParser(description="Spawn tasks for bugbug data pipeline")
+    parser.add_argument("data_pipeline_json")
+
+    args = parser.parse_args()
+    decision_task_id = os.environ.get("TASK_ID")
+    options = get_taskcluster_options()
+    add_self = False
+    if decision_task_id:
+        add_self = True
+        task_group_id = decision_task_id
+    else:
+        task_group_id = taskcluster.utils.slugId()
+    keys = {"taskGroupId": task_group_id}
+
+    id_mapping = {}
+
+    # First pass, do the template rendering and dependencies resolution
+    tasks = []
+
+    with open(args.data_pipeline_json) as pipeline_file:
+        raw_tasks = yaml.load(pipeline_file.read())
+
+    for task in raw_tasks["tasks"]:
+        # Try render the task template
+        context = {}
+        payload = jsone.render(task, context)
+        # raise Exception(payload)
+        # # TODO: the task_id will not match the taskId in the yaml file
```

Comment:
What were you planning to do here?

Reply:
I forget to cleanup the TODO when I introduced the internal ID mapping